File: app/src/main/java/android/academy/nytimes/mvp/RemoteRepository.java
package android.academy.nytimes.mvp;

import android.academy.nytimes.data.NewsItem;

import java.io.IOException;
import java.util.List;

public interface RemoteRepository {

    List<NewsItem> getNews(String category) throws IOException;

    List<String> getCategories();
}

File: app/src/main/java/android/academy/nytimes/mvp/RemoteRepository.kt
package android.academy.nytimes.mvp

import android.academy.nytimes.data.NewsItem
import android.academy.nytimes.utils.Category
import io.reactivex.Single

interface RemoteRepository {

    fun getNews(category: Category): Single<List<NewsItem>>

    fun getCategories(): Array<Category>
}
